collated/test/A0093960X.md

```diff
@@ -0,0 +1,300 @@
+        
+        // type some non undoable command like "list"
+        commandBox.runCommand("list");
+        
+        // undo the delete again
+        assertUndoSuccess(withHoon);
+                
+        // redo the delete
+        assertRedoSuccess(currentList);
+        
+        // delete some random index at 4
+        assertDeleteSuccess(4, currentList);
+        
+        // undo the delete
+        assertUndoSuccess(currentList);
+        
+        // clear the list
+        assertClearCommandSuccess();
+        assertUndoSuccess(currentList);
+        assertRedoSuccess();
+        
+        // add alice
+        TestTask aliceTask = new TestTask(td.alice);
+        assertAddSuccess(aliceTask);
+        
+        // primitive edit undo and redo testing until the assertEditSuccess is complete
+        TestTask aliceTaskBackup = new TestTask(aliceTask);
+        commandBox.runCommand("edit 1 Call Alice from 2pm repeat every day -high");
+        //commandBox.runCommand("edit 1 Call Alice from 2pm to 3pm repeat every day -high");
+        aliceTask.setName(new Name("Call Alice"));
+        aliceTask.setStartDate(DateTime.convertStringToDate("2pm"));
+        //aliceTask.setEndDate(DateTime.convertStringToEndDate("3pm", aliceTask.getStartDate().get()));
+        try {
+            aliceTask.setRecurrence(new RecurrenceRate("1", "day"));
+        } catch (IllegalValueException e) {
+            assert false : "The test data provided cannot be invalid";
+        }
+        aliceTask.setPriority(Priority.HIGH);
+        assertTrue(personListPanel.isListMatching(aliceTask));
+        
+        assertUndoSuccess(aliceTaskBackup);
+        assertRedoSuccess(aliceTask);        
+
+        TestTask recurredAliceTask = new TestTask(aliceTask);
+        recurredAliceTask.setStartDate(DateTime.convertStringToDate("tomorrow 2pm"));
+
+        // primitive done undo and redo testing until the assertDoneSuccess is complete
+        commandBox.runCommand("done 1");
+        assertTrue(personListPanel.isListMatching(recurredAliceTask));
+        commandBox.runCommand("list done");
+        assertTrue(personListPanel.isListMatching(aliceTask));
+        
+        // automatically directs me back to undone view
+        assertUndoSuccess();
+        commandBox.runCommand("list");
+        assertTrue(personListPanel.isListMatching(aliceTask));
+        
+        // automatically directs me back to undone view
+        assertRedoSuccess(recurredAliceTask);
+        commandBox.runCommand("list done");
+        assertTrue(personListPanel.isListMatching(aliceTask));
+
+        commandBox.runCommand("list");
+        
+        // test whether the redo resets properly
+        assertClearCommandSuccess();
+        assertAddSuccess(taskToAdd);
+        
+        
+        // this shows that the redo has reset
+        assertUndoSuccess();
+        assertAddSuccess(aliceTaskBackup);
+        assertRedoSuccess(aliceTaskBackup);
+        
+    }
```

Comment:
i think this test is trying too many things at one place; in general you can apply some SE principles to tests as well; recollect SRP